src/main/java/com/atlassian/db/replica/internal/aurora/AuroraReplicaNode.java

```diff
@@ -1,51 +1,37 @@
 package com.atlassian.db.replica.internal.aurora;
 
-import com.atlassian.db.replica.api.AuroraConnectionDetails;
 import com.atlassian.db.replica.internal.Database;
+import com.atlassian.db.replica.spi.ConnectionPool;
 
 import java.sql.Connection;
-import java.sql.DriverManager;
 import java.sql.SQLException;
-import java.util.Properties;
 import java.util.function.Supplier;
 
 public class AuroraReplicaNode implements Database {
-
-    private final AuroraJdbcUrl auroraUrl;
-    private final AuroraConnectionDetails auroraConnectionDetails;
-    private final ReplicaNode replicaNode = new ReplicaNode();
+    private final String id;
+    private final ConnectionPool connectionPool;
 
     public AuroraReplicaNode(
-        AuroraJdbcUrl auroraUrl,
-        AuroraConnectionDetails auroraConnectionDetails
+        String id,
+        ConnectionPool connectionPool
     ) {
-        this.auroraUrl = auroraUrl;
-        this.auroraConnectionDetails = auroraConnectionDetails;
+        this.id = id;
+        this.connectionPool = connectionPool;
     }
 
     @Override
     public String getId() {
-        return auroraUrl.getEndpoint().getServerId();
+        return id;
     }
 
     @Override
     public Supplier<Connection> getConnectionSupplier() {
         return () -> {
             try {
-                return replicaNode.mark(
-                    getConnection(auroraUrl),
-                    auroraUrl.getEndpoint().getServerId()
-                );
+                return connectionPool.getConnection();
```

Comment:
What about marking connection? It will be done on Jira side? (`replicaNode.mark`)

Reply:
it was removed per @dwatl's suggestion